kernel/kernel-api/src/main/java/io/delta/kernel/ScanFile.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel;
+
+import java.util.Map;
+
+import io.delta.kernel.client.TableClient;
+import io.delta.kernel.data.FilteredColumnarBatch;
+import io.delta.kernel.data.Row;
+import io.delta.kernel.fs.FileStatus;
+import io.delta.kernel.types.*;
+
+import io.delta.kernel.internal.actions.*;
+import io.delta.kernel.internal.fs.Path;
+
+/**
+ * Represents the metadata of {@link FilteredColumnarBatch} returned by
+ * {@link Scan#getScanFiles(TableClient)}. It contains and the schema of the columnar batch
+ * and methods to retrieve specific metadata values (e.x path in `AddFile` Delta Log action)
+ * from each scan file row in the batch.
```

Comment:
This class doc isn't super clear to me

Reply:
Moved this doc to `Scan.getScanFiles()` as it makes more sense to have the schema of of returned scan file columnar batches over there.

This class is no longer needed. Let me know if anything particular is not clear